queries/idl/highlights.scm

```diff
@@ -195,11 +218,104 @@
   (scoped_name
     (identifier) @type))
 
-(simple_declarator
-  (identifier) @attribute)
-
 (array_declarator
   (identifier) @attribute)
 
 (annotation_appl_param
   (identifier) @attribute)
+
+(home_header
+  (identifier) @type)
+
+(factory_dcl
+  (identifier) @type)
+
+(factory_param_dcl
+  "in" @keyword.modifier
+  (type_spec)
+  (simple_declarator) @variable.parameter)
+
+(op_oneway_dcl
+  "oneway" @keyword.modifier
+  (identifier) @function.method)
+
+(in_param_dcl
+  "in" @keyword.modifier
+  (type_spec)
+  (simple_declarator) @variable.parameter)
+
+(context_expr
+  "context" @keyword.modifier)
+
+(get_excep_expr
+  "getraises" @keyword.exception)
+
+(set_excep_expr
+  "setraises" @keyword.exception)
+
+(value_header
+  (identifier) @type)
+
+(value_abs_def
+  (identifier) @type)
+
+(value_forward_dcl
+  (identifier) @type)
+
+(value_box_def
+  (identifier) @type)
+
+(provides_dcl
+  (interface_type) @type
+  (identifier) @variable.member)
+
+(uses_dcl
+  (interface_type)
+  (identifier) @variable.member)
+
+(component_forward_dcl
+  (identifier) @type)
+
+(component_header
+  (identifier) @type)
+
+(porttype_forward_dcl
+  (identifier) @type)
+
+(porttype_def
+  (identifier) @type)
+
+(port_dcl
+  (scoped_name)
+  (identifier) @variable.member)
+
+(connector_header
+  (identifier) @type)
+
+(emits_dcl
+  (scoped_name)
+  (identifier) @variable.member)
+
+(publishes_dcl
+  (scoped_name)
+  (identifier) @variable.member)
+
+(consumes_dcl
+  (scoped_name)
+  (identifier) @variable.member)
```

Comment:
Reading through the parser's code, I believe the `scoped_name` here can be removed without issues. emits_dcl is a sequence of `emits -> (scoped_name) -> (identifier)` so `(emits_dcl (identifier))` should be equivalent to `(emits_dcl (scoped_name) (identifier))`. Same goes for other places.

Reply:
yes, you are right. I removed the useless codes. :)